site/source/docs/api_reference/vr.h.rst

```diff
@@ -0,0 +1,378 @@
+.. _vr-h:
+
+====
+vr.h
+====
+
+The C APIs in `vr.h <https://github.com/kripken/emscripten/blob/master/system/include/emscripten/vr.h>`_ provide basic interfaces for interacting with WebVR from Emscripten.
+
+.. contents:: Table of Contents
+	:local:
+	:depth: 1
+
+.. _test-example-code-vr-api:
+
+Test/Example code
+-----------------
+
+The vr test code demonstrates how to use this API:
+
+	- `test_vr.c <https://github.com/kripken/emscripten/blob/master/tests/test_vr.c>`_
+
+.. _functions-vr-api:
+
+Functions
+---------
+
+Initialization
+==============
+
+.. c:function:: int emscripten_vr_init()
+
+	Initialize the emscripten VR API. This will `navigator.getVRDisplays()
+	<https://w3c.github.io/webvr/spec/1.1/#navigator-getvrdisplays-attribute>`_
+	and when completed, set the return of :c:func:`emscripten_vr_ready` to be `true`.
+
+	:returns: `1` on success, `0` if the browsers WebVR support is insufficient.
+	:rtype: int
+
+.. tip:: This call succeeding is not sufficient for use of the rest of the API. Please
+	make sure to wait until the return value of :c:func:`emscripten_vr_ready` is true.
+
+.. c:function:: int emscripten_vr_ready()
+
+	Check whether the VR API has finished initializing VR displays.
+	See also :c:func:`emscripten_vr_init`.
+
+	This function may return `1` event if WebVR is not supported in the
+	running browser.
+
+	:returns: `1` if ready, `0` otherwise.
+	:rtype: int
+
+API Queries
+===========
+
+All of the following functions require :c:func:`emscripten_vr_init` to have been
+called but do not require VR displays and can therefore be called before the return
+value of :c:func:`emscripten_vr_ready` is `true`.
+
+.. c:function:: int emscripten_vr_version_minor()
+
+	Minor version of the WebVR API currently supported by the browser.
+
+	:returns: minor version of WebVR, or `-1` if not supported or API not initialized.
+	:rtype: int
+
+.. c:function:: int emscripten_vr_version_major()
+
+	Major version of the WebVR API currently supported by the browser.
+
+	:returns: major version of WebVR, or `-1` if not supported or API not initialized.
+	:rtype: int
+
+Display Functions
+=================
+
+All of the following functions require :c:func:`emscripten_vr_init` to have been
+called the return value of :c:func:`emscripten_vr_ready` to be `true`.
+
+.. c:function:: int emscripten_vr_count_displays()
+
+	:returns: Number of displays connected.
+	:rtype: int
+
+.. c:function:: VRDisplayHandle emscripten_vr_get_display_handle(int displayIndex)
+
+	:param int displayIndex: index of display (inclusive 0 to exclusive :c:func:`emscripten_vr_count_displays`).
+	:returns: handle for a VR display.
+	:rtype: VRDisplayHandle
```

Comment:
Here we'll need to be precise what happens if one connects and disconnects displays in different orders, e.g. connect display 1, connect display 2, disconnect display 1, disconnect display 2. Are the `VRDisplayHandle` objects the unique identifiers that are not affected by connects and disconnects, and on connect and disconnect, only the `emscripten_vr_count_displays()` and `emscripten_vr_get_display_handle()` functions are affected?

If one connects a vr display, then disconnects it, and then reconnects the same display, will the `VRDisplayHandle` be the same for that display, or can it be different?

Reply:
The displays will always be present and stay at the same location (in the array), but have their `emscripten_vr_display_connected()` be `true/false` accordingly.
I am usure if a entirely new display is even recognized after `emscripten_vr_init()` at all. I think this would require handling the `"ondisplayconnected"` event.
Will do some research to clarify...

Reply:
`"ondisplayconnected"` passes the connected display to the event handler. This could obviously be a display that already exists in the `WebVR.displays` array. But we currently don't handle this event at all, displays need to therefore be connected before the page is loaded.